compiler/seminst.nim

```diff
@@ -54,7 +54,10 @@ iterator instantiateGenericParamList(c: PContext, n: PNode, pt: TIdTable): PSym
           # later by semAsgn in return type inference scenario
           t = q.typ
         else:
-          localError(c.config, a.info, errCannotInstantiateX % s.name.s)
+          if q.typ.kind == tyCompositeTypeClass:
+            t = errorType(c)
+          else:
+            localError(c.config, a.info, errCannotInstantiateX % s.name.s)
```

Comment:
Wouldn't this be better as
```suggestion
          if q.typ.kind != tyCompositeTypeClass:
            localError(c.config, a.info, errCannotInstantiateX % s.name.s)
```
given the present of `t = errorType(c)` on line [58/61](https://github.com/nim-lang/Nim/pull/20356/files#diff-3006485ebf12b3abba7b715cdf74ef427e3bfdbd943de237d4b005170da8b219L58)